homeassistant/components/vulcan/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The Vulcan component."""
+import logging
+
+from aiohttp import ClientConnectorError
+from vulcan import Account, Keystore, Vulcan
+from vulcan._utils import VulcanAPIException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["calendar"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Uonet+ Vulcan integration."""
+    hass.data.setdefault(DOMAIN, {})
+    try:
+        keystore = Keystore.load(entry.data["keystore"])
+        account = Account.load(entry.data["account"])
+        client = Vulcan(keystore, account)
+        await client.select_student()
+        students = await client.get_students()
+        for student in students:
+            if str(student.pupil.id) == str(entry.data["student_id"]):
+                client.student = student
+                break
+    except VulcanAPIException as err:
+        if str(err) == "The certificate is not authorized.":
+            _LOGGER.error(
```

Comment:
Don't log the error here. Just pass the message as a argument when raising `ConfigEntryAuthFailed`. Otherwise we'll spam the log.